### PR TITLE
[567246] Add a warning message in the DeleteCapellaElement dialog box

### PR DESCRIPTION
**Pull request by lfasani**
This message is displayed when a representation is about to be deleted.

Change-Id: Ib5dd66e0e145d40a4737999c697fc778b2062a44
Signed-off-by: Laurent Fasani <laurent.fasani@obeo.fr>

### CODE REVIEW

**Comment by lfasani** [comment]
-2 because I have to amend this commit to enhance a little the warning message

**Comment by lfasani** [comment]
> -2 because I have to amend this commit to enhance a little the warning message

The code has been updated. It is ready to be reviewed.

**Review by pdulth — CHANGES_REQUESTED** [review]
Message of this dialog is customized by the status returned by CapellaDeleteCommand.preDeleteChecks(). 
https://github.com/eclipse/capella/blob/6ef8e899f3503c655a1c528a2d33442dc722e90d/core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/commands/CapellaDeleteCommand.java#L142-L150

Can you add your warning representation check after the other error message.
In this preCheck method you will have access to getAllElementsToDelete() which will contains the representations instead of _referencingElements

To display a warning instead of an error:
https://github.com/eclipse/capella/blob/6ef8e899f3503c655a1c528a2d33442dc722e90d/core/plugins/org.polarsys.capella.core.ui.toolkit/src/org/polarsys/capella/core/ui/toolkit/dialogs/ConfirmDeleteCapellaElementDialog.java#L286-L289

**Comment by lfasani** [comment]
> 
> 
> Message of this dialog is customized by the status returned by CapellaDeleteCommand.preDeleteChecks().
> 
> https://github.com/eclipse/capella/blob/6ef8e899f3503c655a1c528a2d33442dc722e90d/core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/commands/CapellaDeleteCommand.java#L142-L150
> 
> Can you add your warning representation check after the other error message.
> In this preCheck method you will have access to getAllElementsToDelete() which will contains the representations instead of _referencingElements
> 
> To display a warning instead of an error:
> 
> https://github.com/eclipse/capella/blob/6ef8e899f3503c655a1c528a2d33442dc722e90d/core/plugins/org.polarsys.capella.core.ui.toolkit/src/org/polarsys/capella/core/ui/toolkit/dialogs/ConfirmDeleteCapellaElementDialog.java#L286-L289

comments have been taken into account

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/commands/CapellaDeleteCommand.java [hunk 1]
```diff
@@ -145,11 +151,38 @@ protected IStatus preDeleteChecks() {
...
+            messageNbRepresentation = Messages.CapellaDeleteCommand_ConfirmDeletionWithOneRepresentation;
+          } else {
+            messageNbRepresentation = MessageFormat
+                .format(Messages.CapellaDeleteCommand_ConfirmDeletionWithManyRepresentations, nbRepresentations);
+          }
+          status = Status.WARNING;
+        }
+        String message = MessageFormat.format(Messages.CapellaDeleteCommand_ConfirmDeletionQuestion,
+            messageNbSemanticElement, messageNbRepresentation);
+        protectedElements = new Status(status, CapellaActionsActivator.PLUGIN_ID, message);
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/commands/CapellaDeleteCommand.java:178]: can you rename to "prechecks" or something like that

File: core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/commands/CapellaDeleteCommand.java [hunk 2]
```diff
@@ -145,11 +151,38 @@ protected IStatus preDeleteChecks() {
       if (!CapellaDeleteAction.canDelete(elementsToDelete)) {
         protectedElements = new Status(Status.ERROR, CapellaActionsActivator.PLUGIN_ID,
             Messages.CapellaDeleteCommand_ProtectedElementsError);
+      } else {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.platform.sirius.ui.actions/src/org/polarsys/capella/core/platform/sirius/ui/commands/CapellaDeleteCommand.java:154]: Can you move else outside the first IF

**Review by pdulth — CHANGES_REQUESTED** [review]

File: core/plugins/org.polarsys.capella.core.ui.toolkit/src/org/polarsys/capella/core/ui/toolkit/dialogs/ConfirmDeleteCapellaElementDialog.java [hunk 3]
```diff
@@ -284,12 +285,22 @@ protected Control createContents(Composite parent) {
   private void updateMessageAndButtons() {
     IStatus status = getStatus();
     if (!status.isOK()) {
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.ui.toolkit/src/org/polarsys/capella/core/ui/toolkit/dialogs/ConfirmDeleteCapellaElementDialog.java:287]: I think this shall occurs only on ERROR status.
(if you have a warning, does the OK shall be disabled?)

**Review by pdulth — APPROVED** [review]

**Review by pdulth — APPROVED** [review]

**Pull request merged**